Have git check out main branch if present

zzhengnan opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
Modify git aliases to check out the `main` branch instead of `master` if the former is present.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
To reflect the growing adoption of `main` as the default branch, in place of `master`. 

Currently aliases that rely on `git checkout master` don't work in repos that have adopted this new pattern. With changes in this PR, the modified aliases will behave as follows
- If a `main` branch is present in the repo, check out the `main` branch
- If `master` is present, check out `master`
- If both branches are present, priority is given to `main`

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

I did the following
1. Create two repos, one using `master` as the default branch and another using `main`
2. Create a new branch `dev` in both repos, and check out this branch
3. Run `gcom` and confirm the expected default branch has been checked out in both repos

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [ ] I have added tests to cover my changes, and all the new and existing tests pass.

zzhengnan commented on aliases/available/git.aliases.bash:
```diff
@@ -196,3 +194,17 @@ esac
 }
+
+function gcom() {
+	if git branch | grep -q main; then
```
More context on the `-q` flag:
https://stackoverflow.com/questions/56170215/what-is-the-point-of-grep-q

zzhengnan replied:
grep's man page:
https://www.man7.org/linux/man-pages/man1/grep.1.html

NoahGorny reviewed (changes requested):
Hey @zzhengnan, great idea!

There are many aliases in this file which use hard coded `master` string. Maybe create an helper function `get_default_branch` or something like that?

zzhengnan commented:
@NoahGorny Thanks for the suggestion! Ready for review again. Here's how I tested the code.

Inside a repo whose default branch is `main`.
```
$ git branch
* dev
  main

$ get_default_branch
main

$ gcom
Switched to branch 'main'
```

Inside a repo whose default branch is `master`.
```
$ git branch
* dev
  master

$ get_default_branch
master

$ gcom
Switched to branch 'master'
```

Inside a repo where both `main` and `master` are present.
```
$ git branch
* dev
  main
  master

$ get_default_branch
main

$ gcom
Switched to branch 'main'
```

NoahGorny reviewed (approved):
Well done @zzhengnan !

zzhengnan commented:
@NoahGorny Thanks for the review. Could you help merge the PR?

NoahGorny commented:
> @NoahGorny Thanks for the review. Could you help merge the PR?

I usually wait around for some time to let other contributors chime in, and will merge the PR in the following days :smiley:

zzhengnan commented:
Makes perfect sense. Thanks again.
